packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -930,7 +930,8 @@ export class CrossAccountZoneDelegationRecord extends Construct {
     }
 
     if (!props.delegatedZone.hostedZoneNameServers) {
-      throw Error(`Not able to retrieve Name Servers for ${props.delegatedZone.zoneName} due to it being imported.`);
+      Annotations.of(this).addWarningV2('@aws-cdk/aws-route53:importedDelegatedZoneNSValidation',
+        `Not able to retrieve Name Servers for \'${props.delegatedZone.zoneName}\' due to it being imported. Use a non-imported one or use AwsCustomResource to retrieve the Name Servers.`);
```

Comment:
how do we know this is imported? Not specified doesn't mean that it's been imported.